serde/compat.py

```diff
@@ -194,7 +221,7 @@ def is_bare_list(typ) -> bool:
     >>> is_bare_list(List)
     True
     """
-    return is_list(typ) and typ in (List, list)
+    return typ in (List, list)
```

Comment:
What's the reason for removing `is_list(typ)` 🤔

Reply:
It does not do anything.   
The `typ in (List, list)` check is stricter and determines the output of that statement completely.